source/Visual.cpp

```diff
@@ -24,6 +24,9 @@ using namespace std;
 Visual::Visual(const Effect &effect, Point pos, Point vel, Angle facing, Point hitVelocity)
 	: Body(effect, pos, vel, facing), lifetime(effect.lifetime)
 {
+	if(effect.randomLifetime > 0)
+		lifetime += Random::Int(effect.randomLifetime);
```

Comment:
If the random lifetime is 1, then the lifetime is never increased (since random::int in [0, 1) is always 0). This is why the random lifetime for a projectiles draws from random::int in [0, lifetime + 1).